File: back/src/main/java/com/back/domain/user/dto/TicketSummary.java
package com.back.domain.user.dto;

import com.back.domain.ticket.entity.Ticket;

public record TicketSummary(
        Long ticketId,
        String ticketNumber,
        String seatNumber,
        String gradeName,
        int ticketPrice,
        boolean isValid,
        String createdAt
) {
    public static TicketSummary from(Ticket ticket) {
        return new TicketSummary(
                ticket.getTicketId(),
                ticket.getTicketNumber(),
                ticket.getScheduleSeat().getSeatNumber(),
                ticket.getScheduleSeat().getGradeName(),
                ticket.getTicketPrice(),
                ticket.isValid(),
                ticket.getCreateDate().toLocalDate().toString()
        );
    }
}

File: back/src/main/kotlin/com/back/domain/user/dto/TicketSummary.kt
package com.back.domain.user.dto

import com.back.domain.ticket.entity.Ticket
import com.fasterxml.jackson.annotation.JsonProperty

data class TicketSummary(
    val ticketId: Long,
    val ticketNumber: String,
    val groupToken: String?,
    val seatNumber: String,
    val gradeName: String,
    val ticketPrice: Int,
    @get:JsonProperty("isValid") val isValid: Boolean,
    val createdAt: String
) {
    companion object {
        fun from(ticket: Ticket): TicketSummary = TicketSummary(
            ticketId = checkNotNull(ticket.ticketId) { "Ticket ID null" },
            ticketNumber = ticket.ticketNumber,
            groupToken = ticket.groupToken,
            seatNumber = ticket.scheduleSeat.seatNumber,
            gradeName = ticket.scheduleSeat.gradeName,
            ticketPrice = ticket.ticketPrice,
            isValid = ticket.isValid,
            createdAt = checkNotNull(ticket.createDate) { "Create date null" }.toLocalDate().toString()
        )
    }
}
